tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/HttpClient.scala

```diff
@@ -139,25 +144,25 @@ class HttpClient private (baseUrl: Uri, httpExt: HttpExt)(implicit
   )(implicit um: FromEntityUnmarshaller[A]): IO[Assertion] =
     requestAssert(DELETE, url, None, identity, extraHeaders)(assertResponse)
 
-  def requestAssert[A](
+  def requestAssertAndReturn[A](
       method: HttpMethod,
       url: String,
       body: Option[Json],
       identity: Identity,
       extraHeaders: Seq[HttpHeader] = jsonHeaders
-  )(assertResponse: (A, HttpResponse) => Assertion)(implicit um: FromEntityUnmarshaller[A]): IO[Assertion] = {
+  )(assertResponse: (A, HttpResponse) => (A, Assertion))(implicit um: FromEntityUnmarshaller[A]): IO[(A, Assertion)] = {
```

Comment:
I really dislike our `HttpClient` for reasons similar to why you have done this, but can I suggest an alternative:

Just return `A` and then assert on that in a map or whatever

I guess when the assertion is outside you lose a bit of the debug context but I think our client would generally be more usable if it were working this way. We have some horrible cases where we are running unsafe inside one of these assertion blocks because we want to assert the state of two requests

Reply:
Definitely agree that's the best way, but it would change a lot of places or create some duplication and this PR is already too big... So "for later" 👀